clang/lib/CConv/ConstraintResolver.cpp

```diff
@@ -172,7 +165,7 @@ ConstraintResolver::getTemporaryConstraintVariable(clang::Expr *E,
 // ifc mirrors the inFunctionContext boolean parameter to getVariable.
 std::set<ConstraintVariable *> ConstraintResolver::getExprConstraintVars(
     std::set<ConstraintVariable *> &LHSConstraints, Expr *E,
-    std::set<ConstraintVariable *> &RvalCons, QualType LhsType,
+    QualType LhsType,
     bool &IsAssigned) {
```

Comment:
Why we we need `IsAssiged` flag? Let the callers always use the return value of `getExprConstraintVars`.

Reply:
I think we still need it for the example `x = rand() ? y : f()`, no? This is because `isAssigned` is used in conjunction with `LHSConstraints`, not with `RvalCons`, right?

Reply:
We do not need the flag. We need the assignment..but we can remove the `isAssigned` flag. Basically, I don't see any case where `isAssigned` is true but the return value of `getExprConstraintVars` is non-empty.

Reply:
Good point! I will delete it.